DOC-1752: Document Footnotes Plugin (#2496)
tinymce/tinymce-docs: modules/ROOT/pages/footnotes.adoc

+
Inline-formatting includes setting a string to *Bold* or _Italic_ or `monospaced`.

. Superscripted footnote numbers can be block-formatted
+
If a selected string includes a superscripted footnote number and block-formatting is applied to that string, the superscripted footnote number takes on the applied formatting.
+
Block-formatting includes making a selection a Heading of any level or setting the selection to `Preformatted`.

== Formatting footnote text

. Footnote text can be inline-formatted
+
Footnote text can be formatted using inline-formatting options such as *Bold*, _Italic_, or `monospaced`.

. Footnote text cannot be block-formatted
+
Footnote text cannot, for example, be made into a Heading, or set as `Preformatted`.

== Known issue

Footnote entries are not automatically removed after removing an image caption containing a footnote.

When an image caption with an attached footnote is removed (by unchecking the _Show caption_ checkbox in the **Insert/Edit Image* dialog), the associated footnote text should be automatically removed.

As of this release, however, this automatic removal does not happen.

> The orphan footnote text can be deleted manually. It can also be removed by clicking the `footnotesupdate` button in the toolbar.

Author: I feel like this is more general than specifically image captions. It would be better to specify this as a use case for the `updatefootnotes` button. Seeing as this "bug" occurs from any kind of content removal (of a footnote) directly through the DOM, which is the purpose of the button. Another example of the footnotes section not automatically being updated is when you insert a footnote in a table, and then delete the row/column it exists within; the footnotes section will not automatically update in this case either. Yet another use for the updatefootnotes button.

Reviewer: To quote my commit message:

> Turned known issue re footnotes and image captions into new section on the update footnotes button.

= Footnotes plugin
:navtitle: Footnotes
:description: Insert footnotes into the {productname} editor
:keywords: footnotes, footnote, insertfootnotes, insert, footnotesupdate, mceInsertFootnote, mceUpdateFootnotes
:pluginname: Footnotes
:plugincode: footnotes
:pluginminimumplan: tierthree

include::partial$misc/admon-premium-plugin.adoc[]

include::partial$misc/admon-requires-6.2v.adoc[]

The {pluginname} plugin allows the user to insert a footnote into the {productname} editor.

> Invoking the plugin adds a superscripted footnote number at the current insertion point, or at the end of the current selection.

Reviewer 2: Invoking the plugin sort of makes it sound like that's all it can do. It's just one capability of the plugin. Or is this how we normally write it?

Reviewer: Re-written in response to both comments.

> The plugin then places the insertion point in a new line in the footnotes section located at the bottom of the editor, allowing the user to add the footnote entry related to the newly-created footnote.

Author: @brian-forte might we add more detail to the first line, and move the second two lines below the example?

Reviewer 2: "in a new line in the footnotes section" is a hard to parse. Should we try a different preposition. Maybe 'of' ?

Reviewer: Re-written.

> The plugin then places the insertion point in a new line in the footnotes section located at the bottom of the editor, allowing the user to add the footnote entry related to the newly-created footnote.

Reviewer 2: Is `insertion point` really clearer than `cursor` here?

Reviewer: **cursor**

The UI element that shows where input will be accepted.

At a shell prompt, this has traditionally been a rectangle or underscore glyph, sometimes flashing, sometimes not.

In strings of text in GUI-centric environments (whether the string is being displayed by a text editor or word processor) the cursor is almost always a flashing vertical line.

Many shell prompts have, since the rise of GUIs, also adopted the flashing vertical line as the cursor UI element.

**insertion point**

The place within a document where input will appear.

A cursor can show where the insertion point is, but it is not the insertion point.

It is, absolutely, a fine distinction. But the insertion point is not always shown with a cursor.

**pointer** 

The object on a screen that is controlled by, and acts as the UI proxy for, the mouse (or equivalent pointing device).

Most UI systems change the pointer’s appearance on a contextual basis.

The default arrow pointer is for selecting objects and pressing on-screen buttons.

The I-beam pointer is for selecting text strings and setting the cursor in a text string.

Clicking the I-beam pointer in a text string sets a new insertion point. And the cursor is displayed at this new point to provide visual confirmation of the new insertion point.

And, in the case of the footnotes plugin, the plugin moves the insertion point to a new line in the footnotes section (and the vertical line cursor starts flashing at this new point to show this).

== Interactive Example

liveDemo::footnotes[]

include::partial$misc/purchase-premium-plugins.adoc[]

== Basic setup

[source,js]
----
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  plugins: 'footnotes',
  toolbar: 'footnotes',
});
----

> A footnotes section is automatically created and updated whenever a footnote is inserted into the {productname} editor.

Reviewer 3: Is the footnotes section also updated when a footnote is removed?

Author: ```suggestion
A footnotes section is automatically created or updated whenever a footnote is inserted, reordered, or removed within the {productname} editor.
```
How does this sound?

= Footnotes plugin
:navtitle: Footnotes
:description: Insert footnotes into the {productname} editor
:keywords: footnotes, footnote, insertfootnotes, insert, footnotesupdate, mceInsertFootnote, mceUpdateFootnotes
:pluginname: Footnotes
:plugincode: footnotes
:pluginminimumplan: tierthree

include::partial$misc/admon-premium-plugin.adoc[]

include::partial$misc/admon-requires-6.2v.adoc[]

The {pluginname} plugin allows the user to insert a footnote into the {productname} editor.

> Each time a footnote is added, a superscripted footnote number is placed at the current insertion point, or at the end of the current selection.
> 
> The plugin then adds a new line to the footnotes section located at the bottom of the editor and moves the insertion point to this new line. This allows the user to add the footnote entry related to the newly-created footnote.

Author: ```suggestion
Each time a footnote is added, a superscripted footnote number is placed at the current insertion point, or at the end of the current selection.

The plugin then adds a new line to the footnotes section located at the bottom of the editor and moves the insertion point to this new line. This allows the user to instantly insert content related to the newly-created footnote in the footnote entry.
```
Should we refer to the ` footnote entry` as the location for entering content related to the respective footnote, or as the content itself within the footnotes section?

Reviewer: I’ve pushed a slightly different tweaking of the sentence you re-wrote. Like your re-write it makes the distinction between a footnote entry and its contents more apparent.

It, further, doesn’t leave the inference in place that an end-user is required to add material to a footnote entry.

I don’t have an example to hand, but I can imagine a use-case that automates the creation of footnotes, or the auto-population of footnote entry contents (or both).
